src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java

```diff
@@ -93,6 +93,15 @@ public void validate(QueryState state)
     {
         super.validate(state);
 
+        // Check the length of a valid index name.
+        // Non-valid indexes are validated in IndexMetadata#validate.
+        if (!state.getClientState().isInternal
+            && SchemaConstants.isValidName(indexName, true)
```

Comment:
we do we want to not validate the length for invaild names ?

what about simplifing this check ?

Reply:
Invalid names are always rejected for new and internal index creations in `IndexMetadata.validate`, which can be called for creating indexes without using the statement. Length validation will be added separately as it's more complicated as it should not stop existing indexes with longer names to work.